pkg/udf/forward/forward.go

```diff
@@ -45,6 +44,13 @@ import (
 	"github.com/numaproj/numaflow/pkg/watermark/wmb"
 )
 
+// MapAppliers consolidates the appliers for different map modes together
+type MapAppliers struct {
+	MapUDF       applier.MapApplier
+	MapStreamUDF applier.MapStreamApplier
+	BatchMapUDF  applier.BatchMapApplier
+}
+
```

Comment:
I am not fully in favor of this way of sending appliers to forward. This is because even if we need just `MapUDF`, we still have to set the other `UDFs`. So, I'm thinking, should we use an options approach for all appliers? This way, we just pass the option that we need.

Reply:
Right now, the other appliers would be nil as they are created under the flag. 
If we want to completely do away with the argument and use options I guess that is something we should keep consistent across all other forwards also then?

Reply:
other forwarders only have one UDF and other UDFs like fallback and transformer as passed as options.

Reply:
Sure! If that seems more feasible we can go ahead with that approach. I can do that in a followup PR if that works?
Not to block this further

Reply:
https://github.com/numaproj/numaflow/issues/1809